files/nagios/plugins/check_openstack_interface.py

```diff
@@ -0,0 +1,188 @@
+
+    :returns: credentials, interface name, set IDs/names,
+              set IDs/names to skip, values to filter when using `--all`
+              and check all flag
+    :rtype: Tuple[configparser.ConfigParser, str, set, set, dict, bool]
+    """
+    credentials = ConfigParser()
+    parser = argparse.ArgumentParser("check_openstack_interface")
+    parser.add_argument("interface", type=str, help="interface name")
+    parser.add_argument("-c", "--credential", required=True,
+                        type=argparse.FileType("r"),
+                        help="path to OpenStack credential cnf file")
+    parser.add_argument("--all", action="store_true", help="check all")
+    parser.add_argument("-i", "--id", action="append", type=str, default=[],
+                        help="check specific id/name")
+    parser.add_argument("--skip-id", action="append", type=str, default=[],
+                        help="skip specific id/name")
+    parser.add_argument("--select", action="append", type=str, default=[],
+                        help="use `--select` together with `--all`"
+                             "(e.g. --select subnet=<id>)")
+    args = parser.parse_args()
+
+    if args.interface not in INTERFACE:
+        parser.error("`{}` interface is not supported".format(args.interface))
+
+    if args.all and args.interface in INTERFACE_BY_EXISTENCE:
+        parser.error("flag `--all` is not supported with "
+                     "interface {}".format(args.interface))
+    if args.all and args.id:
+        parser.error("`--all` and `--id` couldn't be given together")
+    elif not args.all and not args.id:
+        parser.error("at least one of '--all/--id' parameters must be entered")
+    elif not args.all and args.skip_id:
+        parser.error("`--skip-id` must be used with `--all`")
+    elif not args.all and args.select:
+        parser.error("`--select` must be used with `--all`")
+
+    credentials.read_file(args.credential)
+
+    return (credentials, args.interface, set(args.id), set(args.skip_id),
+            dict(arg.split("=", 1) for arg in args.select), args.all)
+
+
+def _get_interface(interfaces, name_or_id):
+    """Get interface from list of interfaces.
+
+    :param interfaces: list of OpenStack interfaces
+    :type interfaces: List[Any]
+    :param name_or_id: interface name or ID
+    :type name_or_id: str
+    :returns: interface or None
+    :rtype: Optional[Any]
+    """
+    for interface in interfaces:
+        if (interface.id == name_or_id or
+                (hasattr(interface, "name")) and interface.name == name_or_id):
+            return interface
+
+    return None
+
+
+def _check_interface(name, interfaces, ids):
+    """Check OpenStack interface existence.
+
+    :param name: OpenStack interface name
+    :type name: str
+    :param interfaces: list of OpenStack interfaces
+    :type interfaces: List[Any]
+    :param ids: list of OpenStack interface IDs/names that will be check
```

Comment:
If you enter a name instead of an ID and there are two networks with the same name, you will check both. I don't think it's dangerous and it's the same approach that OpenStack CMD uses.